docs/src/main/asciidoc/using-vertx.adoc

```diff
@@ -0,0 +1,334 @@
+include::./attributes.adoc[]
+= {project-name} - Using Eclipse Vert.x
+
+Eclipse https://vertx.io[Vert.x] is a toolkit for building reactive applications.
+It is designed to be lightweight and embeddable.
+Vert.x defines a reactive execution model and provides a large ecosystem.
+Quarkus integrates Vert.x to implement different forms of reactive.
+Basically, {project-name} uses Vert.x as its reactive engine.
+While lots of reactive features from {project-name} don't _show_ Vert.x, it's used underneath.
+But you can also access the managed Vert.x instance and benefits from the Vert.x ecosystem.
+
+== Installing
+
+To access Vert.x, well, you need to enable the `vertx` extension to use this feature.
+If you are creating a new project, set the `extensions` parameter are follows:
+
+[source, shell, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=vertx-quickstart \
+    -Dextensions="vert.x"
+----
+
+If you have an already created project, the `vertx` extension can be added to an existing {project-name} project with
+the `add-extension` command:
+
+[source,shell]
+----
+mvn quarkus:add-extension -Dextensions="vertx"
+----
+
+Otherwise, you can manually add this to the dependencies section of your `pom.xml` file:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-vertx</artifactId>
+</dependency>
+----
+
+== Accessing Vert.x
+
+Once the extension added, you can access the _managed_ Vert.x instance using `@Inject`:
+
+[source, java]
+----
+@Inject Vertx vertx;
+----
+
+If you are familar with Vert.x, you know that Vert.x provides different API models.
+For instance _bare_ Vert.x uses callbacks, the RX Java 2 version uses `Single`, `Maybe`, `Completable`, `Observable` and `Flowable`.
+
+{project-name} provides 3 Vert.x APIs:
```

Comment:
My initial commit had a paragraph explaining why users would choose one or the others. Why not keeping it? I'm afraid it would be confusing to say there are 3 different APIs without explaining the rationale.

Reply:
I've re-added the paragraph.